CommandLine.c

```diff
@@ -176,6 +180,15 @@ static CommandLineStatus parseArguments(const char* program, int argc, char** ar
                return STATUS_ERROR_EXIT;
             }
             break;
+         case 'n':
+            if (sscanf(optarg, "%16d", &(flags->maxIterations)) == 1) {
+               if (flags->maxIterations < 1)
+                  flags->maxIterations = -1;
```

Comment:
This fails to give a proper error message on invalid numbers put in (like `-n -42`).

Also: the inner `if` should include braces too.

Reply:
> This fails to give a proper error message on invalid numbers put in (like `-n -42`)

Instead of throwing an error message, I thought I would simply set `maxIteraitons` to -1 to "disable" the `-n` option. This is somewhat similar to what's been done for the `--delay` option:

```c
if (sscanf(optarg, "%16d", &(flags->delay)) == 1) {
    if (flags->delay < 1)
        flags->delay = 1;
```

Reply:
> Also: the inner `if` should include braces too.

Sure, I'll fix that.